src/js/RNCSliderNativeComponent.web.js

```diff
@@ -0,0 +1,306 @@
+//    */
+//   onSlidingComplete?: ?(value: number) => void,
+
+//   /**
+//    * Used to locate this view in UI automation tests.
+//    */
+//   testID?: ?string,
+
+//   /**
+//    * Sets an image for the thumb. Only static images are supported.
+//    */
+//   thumbImage?: ?ImageSource,
+
+//   /**
+//    * If true the slider will be inverted.
+//    * Default value is false.
+//    */
+//   inverted?: ?boolean,
+// |}>;
+
+const RCTSliderWebComponent = React.forwardRef(
+  (
+    {
+      value: initialValue,
+      minimumValue = 0,
+      maximumValue = 0,
+      step = 1,
+      minimumTrackTintColor = '#009688',
+      maximumTrackTintColor = '#939393',
+      thumbTintColor = '#009688',
+      thumbStyle = {},
+      style = [],
+      inverted = false,
+      enabled = true,
+      trackHeight = 4,
+      thumbSize = 20,
+      onRNCSliderSlidingStart = () => {},
+      onRNCSliderSlidingComplete = () => {},
+      onRNCSliderValueChange = () => {},
+      ...others
+    },
+    forwardedRef,
+  ) => {
+    const onValueChange = value =>
+      onRNCSliderValueChange &&
+      onRNCSliderValueChange({nativeEvent: {fromUser: true, value}});
+    const onSlidingStart = value =>
+      onRNCSliderSlidingStart &&
+      onRNCSliderSlidingStart({nativeEvent: {fromUser: true, value}});
+    const onSlidingComplete = value =>
+      onRNCSliderSlidingComplete &&
+      onRNCSliderSlidingComplete({nativeEvent: {fromUser: true, value}});
+
+    const containerSize = React.useRef({width: 0, height: 0});
+    const containerRef = forwardedRef || React.createRef();
+    const [value, setValue] = React.useState(initialValue || minimumValue);
+
+    const percentageValue =
+      (value - minimumValue) / (maximumValue - minimumValue);
+    const minPercent = percentageValue;
+    const maxPercent = 1 - percentageValue;
+
+    const containerStyle = StyleSheet.compose(
+      {
+        width: '100%',
+        flexDirection: 'row',
+        userSelect: 'none',
+        alignItems: 'center',
+        cursor: 'pointer',
+      },
+      style,
+    );
+
+    const trackStyle = {
+      height: trackHeight,
+      borderRadius: trackHeight / 2,
+      userSelect: 'none',
+      flexGrow: 1,
+    };
+
+    const minimumTrackStyle = {
+      ...trackStyle,
+      backgroundColor: minimumTrackTintColor,
+      flexBasis: minPercent * 100 + '%',
+    };
+
+    const maximumTrackStyle = {
+      ...trackStyle,
+      backgroundColor: maximumTrackTintColor,
+      flexBasis: maxPercent * 100 + '%',
+    };
+
+    // const width = (containerSize.current ? containerSize.current.width : 0)
+    // const valueOffset = (inverted ? (1 - percentageValue) : percentageValue) * width
+
+    const thumbViewStyle = StyleSheet.compose(
+      {
+        width: thumbSize,
+        height: thumbSize,
+        // left: valueOffset - thumbSize / 2,
+        // top: trackHeight / 2 - thumbSize / 2,
+        // position: absolute,
```

Comment:
I think this should be absolute? it seems to cause a bug for the web... the button width is pushing the maximumTrackStyle view to the right and causing the width of the slider to be rendered incorrectly.

My container is pink and see the slider rendering at 100% width, but outside the cointainer:

![image](https://user-images.githubusercontent.com/1788177/81737876-cae80600-9466-11ea-8c02-5e4ffc8ab160.png)

Now, I am going to delete the DOM element for the thumbView in the Chrome inspector:

![image](https://user-images.githubusercontent.com/1788177/81737996-f23ed300-9466-11ea-9ccb-b2593268b0e6.png)

Reply:
You are right and this was fixed later in a pr that was merged yesterday

Reply:
🙌🏻 wonderful! This is an awesome addition, thank you for doing it